st2api/st2api/controllers/v1/actionalias.py

```diff
@@ -104,80 +97,85 @@ def match(self, action_alias_match_api, **kwargs):
                                                     matches=[],
                                                     command=command)
             return [self._match_tuple_to_dict(match) for match in matches]
-        except (ActionAliasAmbiguityException) as e:
+        except ActionAliasAmbiguityException as e:
             LOG.exception('Command "%s" matched (%s) patterns.', e.command, len(e.matches))
-            pecan.abort(http_client.BAD_REQUEST, str(e))
-            return [self._match_tuple_to_dict(match) for match in e.matches]
+            return abort(http_client.BAD_REQUEST, str(e))
 
-    @request_user_has_permission(permission_type=PermissionType.ACTION_ALIAS_HELP)
-    @jsexpose(arg_types=[str, str, int], body_cls=ActionAliasHelpAPI,
-    status_code=http_client.ACCEPTED)
-    def help(self, action_alias_help_api, **kwargs):
+    def help(self, filter, pack, limit, offset, **kwargs):
```

Comment:
Well, you could use `allOf` construct for parameters too, I believe, so you can create a list of default common params and then extend it with additional ones on a case by case basis. Problem is, at that point we don't really have any indication on the list of filters the particular get_all controller supports so I had to rely on existing tests to infer it. If our coverage is lacking, I might have missed a few. Potentially, quite a few.